vllm/worker/model_runner.py

```diff
@@ -1240,6 +1240,11 @@ def execute_model(
             "finished_requests_ids": model_input.finished_requests_ids,
             "request_ids_to_seq_ids": model_input.request_ids_to_seq_ids,
         } if self.has_seqlen_agnostic else {}
+
+        if (ctrl := model_input.sampling_controller) is not None:
+            assert model_input.sampling_metadata is not None
+            ctrl.prepare(model_input.sampling_metadata)
```

Comment:
This has been changed from RFC (we pass seq group metadata, here, we are using sampling params). Is there any way to hook this with seq group metadata?

Reply:
There are `SamplingParams` in sampling_metadata (specifically in `SequenceGroupToSample`); there is also `request_id` and `sequence_id` - I think this is the stuff you said you needed. The main reason to use sampling metadata is that it also has all the logit positions already computed.

Reply:
I see. yeah I guess what we have in sampling metdata is probably sufficient. let me get back to you soon about this.